service/src/main/java/com/myhome/services/springdatajpa/UserSDJpaService.java

```diff
@@ -63,24 +60,15 @@
   }
 
   @Override
-  public Optional<UserDto> getUserDetails(String userId) {
-    Optional<User> userOptional = userRepository.findByUserIdWithCommunities(userId);
-    return userOptional.map(admin -> {
-      Set<String> communityIds = admin.getCommunities().stream()
-          .map(community -> community.getCommunityId())
-          .collect(Collectors.toSet());
-
-      UserDto userDto = userMapper.userToUserDto(admin);
-      userDto.setCommunityIds(communityIds);
-      return Optional.of(userDto);
-    }).orElse(Optional.empty());
+  public UserDto getUserDetails(String userId) {
+    User user = userRepository.findByUserIdWithCommunities(userId)
+      .orElseThrow(() -> new UserNotFoundException(userId));
+    return userMapper.userToUserDto(user);
   }
 
-  private UserDto createUserInRepository(UserDto request) {
-    User user = userMapper.userDtoToUser(request);
-    User savedUser = userRepository.save(user);
+  private void saveUser(UserDto request) {
```

Comment:
Let's return the `UserDto` from this method and also from the service.
Just ignore it in `UserController`.

That would:
- be easier to tests
- will not have magic logic inside void methods
- will be more functional approach